### PR TITLE
Tests for: magento/magento2#25600, magento/magento2#25528, magento/magento2#25613, magento/magento2#25606.

### PR DESCRIPTION
**Pull request by p-bystritsky**
…

### Description (*)

Test for magento/magento2#25600, magento/magento2#25528, magento/magento2#25613, magento/magento2#25606.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/App/Test/Unit/FrontControllerTest.php [hunk 1]
```diff
@@ -55,7 +87,7 @@ public function testDispatchThrowException()
     {
         $validCounter = 0;
         $callbackValid = function () use (&$validCounter) {
-            return $validCounter++%10 ? false : true;
+            return ($validCounter + 1) % 10 ? false : true;
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/App/Test/Unit/FrontControllerTest.php]: Looks like it will be always false, please double check it

Comment (p-bystritsky) [reply lib/internal/Magento/Framework/App/Test/Unit/FrontControllerTest.php]: @ihor-sviziev thank you. Updated.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6370](https://jira.corp.magento.com/browse/ENGCOM-6370) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**